src/features/Sync/SyncSlice.ts

```diff
@@ -341,111 +443,135 @@ function* updatePassword({ payload: { password } }: UpdatePasswordAction): SagaI
     const DependencyLocator = yield getContext(DEPENDENCY.LOCATOR);
     const syncQueueManager = yield call(DependencyLocator.get, DEPENDENCY.SYNC_QUEUE_MANAGER);
     const syncOutManager = yield call(DependencyLocator.get, DEPENDENCY.SYNC_OUT_MANAGER);
 
     try {
         const syncLogs = yield call(syncQueueManager.nextTemperatureLogs);
-        yield call(syncOutManager.syncTemperatureLogs, syncLogs);
-        yield call(syncTemperatureLogsSuccess);
+        yield call(syncOutManager.syncTemperatureLogs, temperatureLogUrl, syncLogs);
+        yield put(SyncAction.syncTemperatureLogsSuccess(syncLogs));
     } catch (e) {
+        // eslint-disable-next-line no-console
+        console.log(e.message);
         yield call(syncTemperatureLogsFailure);
     }
 }
 
-function* syncTemperatureBreachesSuccess(): SagaIterator {
+function* syncTemperatureBreachesSuccess({ payload: { syncLogs } }: SyncTemperatureBreachesSuccessAction): SagaIterator {
     const DependencyLocator = yield getContext(DEPENDENCY.LOCATOR);
     const syncQueueManager = yield call(DependencyLocator.get, DEPENDENCY.SYNC_QUEUE_MANAGER);
-    yield call(syncQueueManager.dropTemperatureBreaches);
+    yield call(syncQueueManager.dropLogs, syncLogs);
 }
 
 function* syncTemperatureBreachesFailure(): SagaIterator {
     // TODO.
 }
 
-function* syncTemperatureBreaches(): SagaIterator {
+function* syncTemperatureBreaches({ payload: { temperatureBreachUrl }}: SyncTemperatureBreachesAction): SagaIterator {
     const DependencyLocator = yield getContext(DEPENDENCY.LOCATOR);
     const syncQueueManager = yield call(DependencyLocator.get, DEPENDENCY.SYNC_QUEUE_MANAGER);
     const syncOutManager = yield call(DependencyLocator.get, DEPENDENCY.SYNC_OUT_MANAGER);
 
     try {
         const syncLogs = yield call(syncQueueManager.nextTemperatureBreaches);
-        yield call(syncOutManager.syncTemperatureBreaches, syncLogs);
-        yield call(syncTemperatureBreachesSuccess);
+        yield call(syncOutManager.syncTemperatureBreaches, temperatureBreachUrl, syncLogs);
+        yield put(SyncAction.syncTemperatureBreachesSuccess(syncLogs));
     } catch (e) {
-        yield call(syncTemperatureBreachesFailure);
+        // eslint-disable-next-line no-console
+        console.log(e.message);
+        yield put(SyncAction.syncTemperatureBreachesFailure());
     }
 }
 
-function* syncAll(): SagaIterator {
-    yield call(syncSensors);
-    yield call(syncTemperatureLogs);
-    yield call(syncTemperatureBreaches);
+function* syncAll({ payload: { loginUrl, sensorUrl, temperatureLogUrl, temperatureBreachUrl, username, password }}: SyncAllAction): SagaIterator {
+    yield put(SyncAction.authenticate(loginUrl, username, password))
+    yield delay(MILLISECONDS.TEN_SECONDS);
+    yield put(SyncAction.syncSensors(sensorUrl));
+    yield put(SyncAction.syncTemperatureLogs(temperatureLogUrl));
+    yield put(SyncAction.syncTemperatureBreaches(temperatureBreachUrl));
```

Comment:
Hm, how does this work? Is each action fully 'done' before the next?

Reply:
Resolved by #57.